source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -0,0 +1,795 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/geometry_model/ellipsoidal_chunk.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/grid/tria_boundary_lib.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <boost/lexical_cast.hpp>
+
+namespace
+{
+  using namespace dealii;
+  /**
+   * This geometry model implements an (3d) ellipsoidal_chunk geometry which can be non-coordinate parallel.
+   * @author This plugin is a joined effort of Menno Fraters, D Sarah Stamps and Wolfgang Bangerth
+   */
+  namespace Transformations
+  {
+
+    Point<3>
+    EllipsoidalChunkGeometry::pull_back(const Point<3> &space_point) const
+    {
+      return pull_back_ellipsoid (space_point);
+    }
+
+    Point<3>
+    EllipsoidalChunkGeometry::push_forward(const Point<3> &chart_point) const
+    {
+      return push_forward_ellipsoid (chart_point);
+    }
+
+
+    Point<3>
+    EllipsoidalChunkGeometry::push_forward_ellipsoid(const Point<3> &phi_theta_d) const
+    {
+      double R           = aspect::GeometryModel::EllipsoidalChunk<3>::get_semi_major_axis_a();
+      double ellipticity = aspect::GeometryModel::EllipsoidalChunk<3>::get_eccentricity();

```

Comment:
Oh, I see -- wrong class. I think you need to propagate these variables from the `EllipsoidalChunk` class to the `EllipsoidalChunkGeometry` class.